src/services/scheduler.ts

```diff
@@ -147,17 +147,21 @@ export async function start(dbId: string, dbUser: string) {
         }
       },
       rpc: async (payload: any) => {
-        const { params, opid, moduleName, methodName } = payload;
-        let promise;
-        // TODO: look for the method x in module y
+        const { params, opid, modulename, methodname } = payload;
         let output;
         let error;
-        const user = db?.iasqlUsers?.[0];
+        const user = await MetadataRepo.getUserFromDbId(dbId);
         const uid = user?.id;
         const email = user?.email;
-        const dbAlias = db?.alias;
+        const dbAlias = user?.iasqlDatabases?.[0]?.alias;
         try {
-          output = await promise;
+          const versionString = await TypeormWrapper.getVersionString(dbId);
+          const Modules = (modules as any)[versionString];
+          if (!Modules[modulename]) throw new Error(`Module ${modulename} not found`);
+          if (!Modules[modulename][methodname]) {
+            throw new Error(`Method ${methodname} in module ${modulename} not found`);
+          }
+          output = await Modules[modulename][methodname](params);
```

Comment:
Maybe it would make sense to do:

```suggestion
          output = await Modules[modulename][methodname](...params);
```

So arguments can be separately named?

Reply:
Oh, another thing. It may make sense to get a `Context` object here and pass it through, or a custom RPC function that needs to work with the CRUD functions will have to build one itself.

This is what `apply` and `sync` do, but I think the pattern may be common enough that passing in a context object as the first argument could make sense in general?

Reply:
Ahh true!! let me see how I can get it and pass it.

Reply:
Already passing params and context. Not sure if the way I'm passing the context is the right one, though. Please, let me know what you think!